### PR TITLE
feature/vtn-26252 Add optional pluggable search bar into the shared app header

### PR DESCRIPTION
**Pull request by MmtBkn**
https://steel-ventures.atlassian.net/browse/VTN-26252

### CODE REVIEW

**Comment by SteveShaffer** [comment]
@MmtBkn if you can get a code review and approval from someone else like @KhuongVeritone or a member of essentials team, I'll approve (since it seems the codeowners lock is gonna stay on this PR even though I turned it off in settings).

File: packages/veritone-widgets/src/redux/configureStore.js [hunk 1]
```diff
@@ -26,16 +26,62 @@ const enhancer = composeEnhancers(
   )
 );
 
+function createReducer(asyncReducers) {
```
Comment (MmtBkn) [inline packages/veritone-widgets/src/redux/configureStore.js]: @kenakaiser we can remove this function

File: packages/veritone-widgets/src/redux/configureStore.js [hunk 2]
```diff
@@ -26,16 +26,62 @@ const enhancer = composeEnhancers(
...
     ...createStore(createRootReducer(), initialState, enhancer),
     asyncReducers: {},
     runSaga: sagaMiddleware.run,
     sagaMiddleware
-    // close: () => store.dispatch(END);
   };
 
+  // Create an inject reducer function
+  // This function adds the async reducer, and creates a new combined reducer
+  store.injectReducer = (key, asyncReducer) => {
```
Comment (MmtBkn) [inline packages/veritone-widgets/src/redux/configureStore.js]: @kenakaiser we can also remove this

File: packages/veritone-widgets/src/redux/configureStore.js [hunk 3]
```diff
@@ -26,16 +26,62 @@ const enhancer = composeEnhancers(
...
+
+  // Add injectSaga method to our store
+  store.injectSaga = createSagaInjector(sagaMiddleware.run, rootSaga);
+
   store.runSaga(rootSaga);
 
   return store;
 }
+
+export { rootSaga, reducers };
```
Comment (MmtBkn) [inline packages/veritone-widgets/src/redux/configureStore.js]: @kenakaiser I believe we only need this change here

**Pull request merged**